Automattic/jetpack
# My Jetpack: Update data handling - Implement request status in Product Card

Fixes #

#### Changes proposed in this Pull Request:


This PR updates the data handling implementation for the Products;

* Change the response shape for the `/site/products/$product` endpoint
* Update Redux implementation according to it
* Implement the initial requesting status for the Product card when an action is run

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Go to 'My Jetpack dashboard
* Play with the Backup card. You should see the error when trying to deactivate the product. It's expected since the handler is not implemented yet on the server-side.
* You can see how the data is propagated using network and redux tool of your dev tools, in the client:


https://user-images.githubusercontent.com/77539/150874627-2e734bfd-d910-4db8-bce2-aaad9f36a4c0.mov

You can hardcode the response for both for the `activate_product()` and `detivate_product()` methods respectively

```php
$deactivate_product_result = $product;
$deactivate_product_result['status'] = 'active';
return rest_ensure_response( $deactivate_product_result, 200 );
```

```php
$deactivate_product_result = $product;
$deactivate_product_result['status'] = 'inactive';
return rest_ensure_response( $deactivate_product_result, 200 );
```

for the `detivate_product()` method


https://user-images.githubusercontent.com/77539/150891154-64500c82-7b2f-4fb4-92e5-67a7ed985c1f.mov
